docs/gcloud-auth.rst

```diff
@@ -0,0 +1,208 @@
+.. toctree::
+  :maxdepth: 1
+  :hidden:
+
+Authentication
+--------------
+
+=======
+Summary
+=======
+
+For the majority of cases,
```

Comment:
"For the majority of cases" doesn't help me determine whether I should rely on the library or not.

Can you add details as appendix like:

```
For the majority of cases (see [details](link to the appendix) for exceptions),
```

```
When you should create credentials on your own.
* blah
* blah
```